src/analyzer/plugins/analyzerkeyfinder.cpp

```diff
@@ -0,0 +1,115 @@
+#include "analyzer/plugins/analyzerkeyfinder.h"
+
+#include "analyzer/constants.h"
+#include "util/assert.h"
+#include "util/math.h"
+
+using mixxx::track::io::key::ChromaticKey;
+using mixxx::track::io::key::ChromaticKey_IsValid;
+
+namespace {
+ChromaticKey chromaticKeyFromKeyFinderKeyT(KeyFinder::key_t key) {
+    switch (key) {
+    case (KeyFinder::A_MAJOR):
+        return ChromaticKey::A_MAJOR;
+    case (KeyFinder::A_MINOR):
+        return ChromaticKey::A_MINOR;
+    case (KeyFinder::B_FLAT_MAJOR):
+        return ChromaticKey::B_FLAT_MAJOR;
+    case (KeyFinder::B_FLAT_MINOR):
+        return ChromaticKey::B_FLAT_MINOR;
+    case (KeyFinder::B_MAJOR):
+        return ChromaticKey::B_MAJOR;
+    case (KeyFinder::B_MINOR):
+        return ChromaticKey::B_MINOR;
+    case (KeyFinder::C_MAJOR):
+        return ChromaticKey::C_MAJOR;
+    case (KeyFinder::C_MINOR):
+        return ChromaticKey::C_MINOR;
+    case (KeyFinder::D_FLAT_MAJOR):
+        return ChromaticKey::D_FLAT_MAJOR;
+    case (KeyFinder::D_FLAT_MINOR):
+        return ChromaticKey::C_SHARP_MINOR;
+    case (KeyFinder::D_MAJOR):
+        return ChromaticKey::D_MAJOR;
+    case (KeyFinder::D_MINOR):
+        return ChromaticKey::D_MINOR;
+    case (KeyFinder::E_FLAT_MAJOR):
+        return ChromaticKey::E_FLAT_MAJOR;
+    case (KeyFinder::E_FLAT_MINOR):
+        return ChromaticKey::E_FLAT_MINOR;
+    case (KeyFinder::E_MINOR):
+        return ChromaticKey::E_MINOR;
+    case (KeyFinder::F_MAJOR):
+        return ChromaticKey::F_MAJOR;
+    case (KeyFinder::F_MINOR):
+        return ChromaticKey::F_MINOR;
+    case (KeyFinder::G_FLAT_MAJOR):
+        return ChromaticKey::F_SHARP_MAJOR;
+    case (KeyFinder::G_FLAT_MINOR):
+        return ChromaticKey::F_SHARP_MINOR;
+    case (KeyFinder::G_MAJOR):
+        return ChromaticKey::G_MAJOR;
+    case (KeyFinder::G_MINOR):
+        return ChromaticKey::G_MINOR;
+    case (KeyFinder::A_FLAT_MAJOR):
+        return ChromaticKey::A_FLAT_MAJOR;
+    case (KeyFinder::A_FLAT_MINOR):
+        return ChromaticKey::G_SHARP_MINOR;
+    case (KeyFinder::SILENCE):
+    default:
+        return ChromaticKey::INVALID;
+    }
+}
+} // anonymous namespace
+
+namespace mixxx {
+
+AnalyzerKeyFinder::AnalyzerKeyFinder()
+        : m_keyFinder(),
```

Comment:
Explicitly invoking the default constructor of members is uncommon. This is the default behavior.

Reply:
fixed 49a0948